### PR TITLE
MTSF10: Implemented add and view theatre section

### PR DESCRIPTION
**Pull request by Tshering-Tee**
## Description
Implemented "View Theatres" page with data fetched from the database, added live search by name and location, and an "Add" button to open a dialog box.

## Changes made

- [x] Created "View Theatres" page to display theatre data from the database.
- [x] Implemented live search functionality by name and location fields.
- [x] Added an "Add" button to the page.
- [x] Configured a dialog box to open when the "Add" button is clicked.

## Dependencies added

- [x] Angular Materials Installed (ng add @angular/material)

## Screenshots 

image

image

### CODE REVIEW

**Review by rigyel-dev — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -43,4 +45,4 @@
     "tailwindcss": "^3.4.16",
     "typescript": "~5.6.2"
   }
```
Comment (sonam-yoezer) [inline package.json]: add a line break

File: src/app/admin/theatre/add-theatre-dialog/add-theatre-dialog.component.css [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    }
+    to {
+      opacity: 0;
+      transform: scale(0.95);
+    }
+  }
+  
+  .dialog-exit {
+    animation: slideOut 0.2s ease-in forwards;
+  }
```
Comment (sonam-yoezer) [inline src/app/admin/theatre/add-theatre-dialog/add-theatre-dialog.component.css]: add a a line break

File: src/app/admin/theatre/add-theatre-dialog/add-theatre-dialog.component.html [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+            <button
+                type="submit"
+                [disabled]="loading || theatreForm.invalid"
+                class="flex items-center px-6 py-2 bg-[#F1B434] text-[#1B355E] rounded-lg hover:bg-[#1B355E] hover:text-[#F1B434] transition-all duration-300 disabled:opacity-50 disabled:cursor-not-allowed transform hover:scale-105">
+                <span class="material-icons mr-2">add_circle</span>
+                <span>{{ loading ? 'Adding...' : 'Add Theatre' }}</span>
+            </button>
+        </div>
+    </form>
+</div>
```
Comment (sonam-yoezer) [inline src/app/admin/theatre/add-theatre-dialog/add-theatre-dialog.component.html]: add a line break

File: src/app/admin/theatre/add-theatre-dialog/add-theatre-dialog.component.ts [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    if (dialogContainer) {
+      dialogContainer.classList.add('dialog-exit');
+      setTimeout(() => {
+        this.dialogRef.close();
+      }, 200);
+    } else {
+      this.dialogRef.close();
+    }
+  }
+}
```
Comment (sonam-yoezer) [inline src/app/admin/theatre/add-theatre-dialog/add-theatre-dialog.component.ts]: add line break

File: src/app/admin/theatre/view-theatre/view-theatre.component.css [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  
+  .table-container::-webkit-scrollbar-thumb {
+    background: #F1B434;
+    border-radius: 4px;
+  }
+  
+  .table-container::-webkit-scrollbar-thumb:hover {
+    background: #1B355E;
+  }
+
```
Comment (sonam-yoezer) [inline src/app/admin/theatre/view-theatre/view-theatre.component.css:31]: add line break

File: src/app/admin/theatre/view-theatre/view-theatre.component.html [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+          </table>
+        </div>
+      </div>
+  
+      <!-- Empty State -->
+      <div *ngIf="!loading && theatres.length === 0" class="text-center py-8">
+        <div class="text-gray-500">No Theatres found</div>
+      </div>
+    </div>
+  </div>
```
Comment (sonam-yoezer) [inline src/app/admin/theatre/view-theatre/view-theatre.component.html]: add line break

**Review by sonam-yoezer — APPROVED** [review]

**Pull request merged**